contracts/Presale.sol

```diff
@@ -223,21 +242,24 @@ contract Presale is IPresale, Ownable2Step, ReentrancyGuard, Pausable {
             );
         }
 
-        require(_totalAllocation > 0, "MIN_ALLOCATION");
+        uint256 _finalRoundIndex = _rounds.length - 1;
```

Comment:
just revisiting this. 
Lets create a `uint256 _totalRounds` and delete the `RoundConfig[] memory` so that we are not loading the entire array into storage. 

Lets just load the `RoundConfig memory _round = $rounds[i];` each iteration.

Reply:
added `_totalRounds` instead of loading all rounds into memory on each `_sync`